.github/workflows/main.yml

```diff
@@ -111,15 +114,17 @@ jobs:
                   workspace=$(jq -r '.name' packages/${{ matrix.dir }}/package.json)
                   yarn workspace "$workspace" run test:coverage
 
-    coveralls:
-        runs-on: ubuntu-latest
-        if: needs.changed-files.outputs.any_changed == 'true' && github.event_name == 'push' && github.ref == 'refs/heads/main'
-        needs: [changed-files, tests]
-        steps:
-            - name: Coveralls
+            - if: contains(needs.changed-files.outputs.modified_files, matrix.dir) && github.event_name == 'push' && github.ref == 'refs/heads/main'
+              name: Coveralls
               uses: coverallsapp/github-action@v2
               with:
                   github-token: ${{ secrets.GITHUB_TOKEN }}
+                  parallel: true
+                  flag-name: run ${{ join(matrix.*, '-') }}
+
+            # this is dummy step was a workaround against the GH limitation of having ci checks to be required even if not run
+            # see https://github.com/orgs/community/discussions/13690
+            - if: needs.changed-files.outputs.any_changed != 'true'
```

Comment:
a bit hacky
But otherwise all the tests ci required check will be stuck in "waiting for status to be reported" in the case where the tests job was skipped